universe/vault/credentials.go

```diff
@@ -48,19 +63,70 @@ func (c *Credentials) Login(ctx context.Context, options ...vault.ClientOption)
 		}
 	}
 
-	resp, err := client.Auth.AppRoleLogin(
+	return &ClientHandle{
+		creds:  c,
+		client: client,
+	}, nil
+}
+
+func (h *ClientHandle) Get(ctx context.Context) (*vault.Client, error) {
+	h.m.Lock()
+	defer h.m.Unlock()
+
+	// Re-authenticate if less than six seconds left.
```

Comment:
Works, but a bit harder to parse.
We usually prefer simple return guards, e.g. 

```
	if time.Until(h.expires()) > ttlBuffer {
		return h.client, nil
	}

	if err := h.renew(ctx); err != nil {
		return nil, err
	}

	return h.client, nil
```

Also, I'd skip the comment (it is already outdated).

Reply:
(I thought I removed that comment already.)

Fair point about early-return, plus it is nice to always easier to follow if one of (ptr, err) is a literal nil.